app/models/game.rb

```diff
@@ -36,23 +36,22 @@ def determine_checkmate
     false
   end
 
-  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Style/ParallelAssignment, Metrics/MethodLength
   def checkmate(id)
     current_pieces = pieces.where(user_id: id, captured: false)
-    check_status = true
+    status = true
     current_pieces.each do |piece|
       8.times do |row|
         8.times do |col|
           next unless piece.valid_move?(row, col)
-          Piece.transaction do
-            piece.try_to_move(row, col)
-            check_status = false if determine_check == false
-            fail ActiveRecord::Rollback
-          end
+          orig_row, orig_col = piece.row_position, piece.col_position
+          status = false unless checkmate_check(piece, row, col)
+          # If transaction moves piece, move it back to original position
+          piece.update_attributes(row_position: orig_row, col_position: orig_col)
```

Comment:
This might actually potentially cause bugs since for pawns, we are storing the past updates using ActiveRecord Dirty. If we use transactions, everything is rolled back so this problem doesn't exist.

I figured out why transactions wasn't working - we simply had to reload the pieces with `.reload` I think